src/org/labkey/targetedms/chromlib/LibSampleFileDao.java

```diff
@@ -60,7 +60,15 @@ protected void setValuesInStatement(LibSampleFile sampleFile, PreparedStatement
         }
         stmt.setString(colIndex++, sampleFile.getInstrumentIonizationType());
         stmt.setString(colIndex++, sampleFile.getInstrumentAnalyzer());
-        stmt.setString(colIndex, sampleFile.getInstrumentDetector());
+        stmt.setString(colIndex++, sampleFile.getInstrumentDetector());
+        if (sampleFile.getCePredictorId() != null)
+        {
+            stmt.setInt(colIndex++, sampleFile.getCePredictorId());
```

Comment:
We need to always set these, or the column indices will get confused. Use setObject() to account for the null values.